elodie/filesystem.py

```diff
@@ -378,12 +389,22 @@ def process_file(self, _file, destination, media, **kwargs):
         if(move is True):
             stat = os.stat(_file)
             shutil.move(_file, dest_path)
+            if(os.path.exists(_file + '_original')):
+                os.remove(_file + '_original')
             os.utime(dest_path, (stat.st_atime, stat.st_mtime))
         else:
-            compatability._copyfile(_file, dest_path)
+            # exiftool preserves the original file with '_original' appended to
+            # the file name.
+            if(os.path.exists(_file + '_original')):
+                shutil.move(_file, dest_path)
+                shutil.move(_file + '_original', _file)
+            else:
+                compatability._copyfile(_file, dest_path)
             self.set_utime_from_metadata(media.get_metadata(), dest_path)
 
-        db.add_hash(checksum, dest_path)
+        db = Db()
+        db.add_hash(original_checksum, dest_path)
+        db.add_hash(new_checksum, dest_path)
         db.update_hash_db()
```

Comment:
Why are both checksums being added to the db? Should we not remove `original_checksum` instead?

Reply:
I updated the PR with the requested changes, but noticed there is still an issue where the modified timestamp on the original file has been updated and is not the original timestamp. Was looking into fixing this but haven't got it working yet. Do not merge for now.

Reply:
In terms of the two checksums being added. A photo can have a checksum against the source photo where no metadata has been modified, and also a checksum against the photo in the destination where metadata may have been modified.

So it seems we need to cater for two cases:
- If the user imports a photo from a source, then tries to reimport that photo from the same source. Elodie should not attempt to reimport this, and it won't as we can check against the original hash (as it is added to the db).
- If the user imports a photo, then for some reason copies the destination photo with new tags to a different location and tries to reimport that again. Elodie would then not reimport this as the new checksum already exists.

Does this make sense? I realise it is at the expense of doubling the size of the db.json. But not sure on how we could do it otherwise.

Reply:
As for the timestamp of the original file's timestamp I think the only way to preserve that is to store it before exiftool modifies it and then set the time once the `_original` file is moved back to source path.

I did look online and found [this site](http://webcache.googleusercontent.com/search?q=cache:pRofGFwx9q4J:https://wtanaka.com/node/7701&num=1&client=firefox-b-1-ab&hl=en&gl=us&strip=1&vwsrc=0) which states that the `_original` file retains the timestamp.

Reply:
Hi, sorry had a few things on! Yes, I read your comments and I agree.

Maybe I didn't explain the second case fully however, just wanted to rectify that.

> If the user imports a photo, then for some reason copies the destination photo with new tags to a different location and tries to reimport that again. Elodie would then not reimport this as the new checksum already exists in the db.

What I actually mean is:
If the user imports a photo, Elodie creates a new file in the destination with updated tags in comparison to the original (the user is not manually adjusting the tags). If the user then copies the destination photo to a different location and tries to reimport again, Elodie should not reimport this as the new checksum already exists in the db. 

The difference being that the user is not manually modifying the tags on the destination themselves, this seems to be done by Elodie on the test files I had. But if the user tries to reimport that from somewhere else, they could end up with two in the destination library if we don't store the second hash. I will double check the actual behaviour without storing the second hash as am not certain (a bit of time has passed so will check this over). Maybe it's a non issue.

With the timestamp, yes, I will have a look.

Reply:
> If the user imports a photo, Elodie creates a new file in the destination with updated tags in comparison to the original (the user is not manually adjusting the tags). If the user then copies the destination photo to a different location and tries to reimport again, Elodie should not reimport this as the new checksum already exists in the db.

When deciding if a file exists there are two things which are verified. See the `if/else` statement [here](https://github.com/jmathai/elodie/blob/master/elodie/filesystem.py#L358).

1. Does the checksum of the file exist in `hash.json`?
2. Does the file exist at the expected location specified in `hash.json`?

We can perform both checks safely since a photo's target destination is deterministic.

```
fn(photo, metadata) = location
```

If a file is manually moved then Elodie is not aware of it. An attempt to reimport that file will find a checksum match but fail to find the file at the expected location and import it again. The moved copy will remain untracked by Elodie.

Hope that makes sense. Let me know if I misunderstood your point.